.github/composite_actions/log_metric/action.yaml

```diff
@@ -0,0 +1,52 @@
+name: 'Log Metric'
+description: 'Log data point to a metric with the provided value. If the metric is not there, it will create one.'
+inputs:
+  aws-region:
+    required: true
+    description: The AWS region
+  role-to-assume:
+    required: true
+    description: The role to assume in the STS session
+  metric-name:
+    description: 'target metric name'
+    required: true
+  value:
+    description: 'the value to log to the metric'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@67fbcbb121271f7775d2e7715933280b06314838 # 1.7.0
+      with:
+        role-to-assume: ${{ inputs.role-to-assume }}
+        aws-region: ${{ inputs.aws-region }}
+        role-duration-seconds: 900
+
+    - name: Create AWS profile
```

Comment:
I think this can be removed here as creating the profile only needed for running amplify pull in headless env (as CLI doesn't accept token). For calls to aws sdk, the creds available as env variables should be sufficient.

Reply:
thought about this more, I guess not a blocker here as we could add a bool param to this action to skip getting creds and could set that for some other consumer that already has creds if/when needed in future